theano/tensor/basic.py

```diff
@@ -4607,7 +4607,7 @@ def reshape(x, newshape, ndim=None, name=None):
                 "to know what the number of dimensions of the reshaped "
                 "variable will be. You can provide the 'ndim' keyword "
                 "argument to 'reshape' to avoid this problem." % newshape)
-    op = Reshape(ndim, name)
```

Comment:
You need to modify the class Reshape to assert that the name it receive is None and don't set it as an attribute.